### PR TITLE
Bugfix for nondeterminstic torch_key

### PR DESCRIPTION
**Pull request by jansel**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #128111

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @desertfire @chauhang

### CODE REVIEW

File: torch/_inductor/codecache.py [hunk 1]
```diff
@@ -564,9 +564,13 @@ def get_code_hash(roots):
...
         with open(module, "rb") as f:
-            contents[module] = f.read()
-
-    return hashlib.sha256(pickle.dumps(contents)).digest()
+            contents[spec.name] = f.read()
+    hasher = hashlib.sha256()
+    # Iterate over dict in sorted order since iter_modules may not be deterministic
+    for name in sorted(contents.keys()):
+        hasher.update(name.encode("utf-8"))
+        hasher.update(contents[name])
```
Comment (Skylion007) [inline torch/_inductor/codecache.py:572]:
```suggestion
    for name, content in sorted(contents.items()):
        hasher.update(name.encode("utf-8"))
        hasher.update(content)

**Review by Skylion007 — COMMENTED** [review]
nit

File: torch/_inductor/codecache.py [hunk 2]
```diff
@@ -656,6 +660,7 @@ def __init__(
         self.torch_version = torch_key()
```
Comment (oulgen) [inline torch/_inductor/codecache.py:660]: torch_key is already here

File: torch/_inductor/codecache.py [hunk 3]
```diff
@@ -656,6 +660,7 @@ def __init__(
...
         self.system_info = CacheBase.get_system()
         self.inductor_config = config.save_config_portable()
+        self.torch_key = torch_key()
```
Comment (oulgen) [inline torch/_inductor/codecache.py:663]: delete this

**Review by oulgen — COMMENTED** [review]
since torch_key was already below, I'm not really sure what this is fixing?

I agree that instead of pickle, iterating is better but functionally this should be a no-op?

**Review by oulgen — APPROVED** [review]

**Pull request closed without merging**